docs/velocity/dev/api/plugin-messaging.mdx

```diff
@@ -17,6 +17,20 @@ flowchart LR
     backend -->|"3 (Incoming)"| Velocity -->|"4 (Outgoing)"| player
 ```
 
+:::warning
+
+When listening to PluginMessageEvent, ensure sure the result is
+<Javadoc name={"com/velocitypowered/api/event/connection/PluginMessageEvent$ForwardResult#handled()"} project={"velocity"}>`ForwardResult.handled()`</Javadoc>
+if you do not intend the client to participate.
+
+If the result is forwarded, Players can spoof the proxy to your backends.
+
+Additionally, ensure the result is set correct after actually handling correct messages, to prevent them from being leaked to the other party.
```

Comment:
```suggestion
Additionally, ensure the result is set correctly after actually handling correct messages, to prevent them from being leaked to the other party.
```